x-pack/plugins/apm/server/lib/apm_telemetry/collect_data_telemetry/tasks.test.ts

```diff
@@ -376,74 +377,223 @@ describe('data telemetry collection tasks', () => {
+                    {
+                      key: '10m',
+                      doc_count: 1080,
+                      metrics_value_count: {
+                        value: 6,
+                      },
+                    },
+                    {
+                      key: '1m',
+                      doc_count: 1080,
+                      metrics_value_count: {
+                        value: 6,
+                      },
+                    },
+                    {
+                      key: '60m',
+                      doc_count: 1080,
+                      metrics_value_count: {
+                        value: 6,
+                      },
+                    },
+                  ],
+                },
+              },
+              {
+                key: 'service_destination',
+                doc_count: 1620,
+                rollup_interval: {
+                  buckets: [
+                    {
+                      key: '10m',
+                      doc_count: 540,
+                      metrics_value_count: {
+                        value: 3,
+                      },
+                    },
+                    {
+                      key: '1m',
+                      doc_count: 540,
+                      metrics_value_count: {
+                        value: 3,
+                      },
+                    },
+                    {
+                      key: '60m',
+                      doc_count: 540,
+                      metrics_value_count: {
+                        value: 3,
+                      },
+                    },
+                  ],
+                },
+              },
+              {
+                key: 'service_summary',
+                doc_count: 30,
+                rollup_interval: {
+                  buckets: [
+                    {
+                      key: '1m',
+                      doc_count: 12,
+                      metrics_value_count: {
+                        value: 12,
+                      },
+                    },
+                    {
+                      key: '10m',
+                      doc_count: 9,
+                      metrics_value_count: {
+                        value: 9,
+                      },
+                    },
+                    {
+                      key: '60m',
+                      doc_count: 9,
+                      metrics_value_count: {
+                        value: 9,
+                      },
+                    },
+                  ],
+                },
+              },
+              {
+                key: 'span_breakdown',
+                doc_count: 12,
+                rollup_interval: {
+                  buckets: [],
+                },
+              },
+              {
+                key: 'app',
+                doc_count: 6,
+                rollup_interval: {
+                  buckets: [],
+                },
+              },
+            ],
+          },
+        },
+      };
```

Comment:
Shouldn't this be an API test?

Reply:
So currently all the tasks can be executed using 1 endpoint which is the `internal/apm/debug-telemetry`. But then it will execute all the tasks and that is something we should aim also. But since it was not present, i refactored the existing Unit test to at least protect us with respect to logic changes